// @cucumber/pretty-formatter/dist/src/TextBuilder.js
import { styleText } from 'node:util';
export class TextBuilder {
    stream;
    supportsColor;
    text = '';
    constructor(stream) {
        this.stream = stream;
        this.supportsColor = styleText('green', '-', { stream }) !== '-';
    }
    applyStyle(value, style) {
        if (!style) {
            return value;
        }
        if (style === 'default') {
            // util.styleText doesn't support 'default' style
            return this.supportsColor ? `\u001b[39m${value}\u001b[0m` : value;
        }
        return styleText(style, value, { stream: this.stream });
    }
    space() {
        this.text += ' ';
        return this;
    }
    line() {
        this.text += '\n';
        return this;
    }
    append(value, style) {
        this.text += this.applyStyle(value.toString(), style);
        return this;
    }
    build(style, styleEachLine = false) {
        if (styleEachLine) {
            return this.text
                .split('\n')
                .map((line) => this.applyStyle(line, style))
                .join('\n');
        }
        return this.applyStyle(this.text, style);
    }
}
